tasks/ast_codegen/src/main.rs

```diff
@@ -129,7 +125,11 @@ impl CodegenCtx {
                 ident_table.insert(ident, type_id);
             }
         }
-        Self { modules: mods, ty_table, ident_table }
+
+        let mut me = Self { ty_table, ident_table, schema: Vec::default() }.link(linker)?;
+        let schema = mods.into_iter().map(Module::build).collect::<Result<Vec<_>>>()?;
+        _ = std::mem::replace(&mut me.schema, schema);
```

Comment:
Is there any reason why this doesn't work?

```suggestion
        me.schema = schema;
```

Reply:
Nevermind. It's replaced in next PR in the stack anyway.